### PR TITLE
SSProto 2

### PR DESCRIPTION
**Pull request by Hexawolf**
New protocol version with removed ed448 signatures.

### CODE REVIEW

**Comment by Hexawolf** [comment]
Oh, sure! Also, the client may do a self-update in case if the protocol version is bad. I'll update PR name until everything gets fixed!

File: ss-client/build.sh [hunk 1]
```diff
@@ -1,14 +1,13 @@
 #!/bin/sh
 
-if [ $# -ne 4 ]; then
-    echo "Usage: ./build.sh KEY-FILE CERTIFICATE SERVER-ADDRESS FILENAME"
-    echo "E.g. ./build.sh ss.key cert.pem doggoat.de Updater"
+if [ $# -ne 3 ]; then
+    echo "Usage: ./build.sh CERTIFICATE SERVER-ADDRESS FILENAME"
+    echo "E.g. ./build.sh cert.pem doggoat.de Updater"
```
Comment (foxcpp) [inline ss-client/build.sh]: Somewhere else you changed targetHost to include port. build.sh usage example is not updated to account for this change.

```suggestion
    echo "E.g. ./build.sh cert.pem doggoat.de:48879 Updater"
```

File: ss-client/main.go [hunk 2]
```diff
@@ -100,9 +100,35 @@ func Crash(data ...interface{}) {
...
+	}
+	fi, err := os.Stat(p)
+	if err == nil {
+		if !fi.Mode().IsDir() {
+			return p, nil
+		}
+		err = fmt.Errorf("%s is directory", p)
+	}
+	if filepath.Ext(p) == "" {
+		p += ".exe"
```
Comment (foxcpp) [inline ss-client/main.go]: What's about non-Windows platforms?

File: ss-client/main.go [hunk 3]
```diff
@@ -191,39 +217,57 @@ SOFTWARE.`)
 		}
 	}
 	fmt.Println("Default directory for installation is", installDirectory)
-	os.MkdirAll(installDirectory+"mods", 0770)
+	os.MkdirAll(installDirectory+"mods", 0775)
 	os.MkdirAll(installDirectory+"config", 0770)
-	os.MkdirAll(installDirectory+"versions", 0770)
+	os.MkdirAll(installDirectory+"versions", 0775)
```
Comment (foxcpp) [inline ss-client/main.go]: Side note: It makes very little sense to set certain mode here. Let user handle it by setting umask (default is enough tho).
```
effective_mode = this_mode & ~umask
```

File: ss-client/main.go (hunk 2 above)
Comment (foxcpp) [inline ss-client/main.go]: What's about non-Windows platforms?

Comment (Hexawolf) [reply ss-client/main.go]: Not gonna lie - this code is nearly a copy-paste from another project. Of course, this won't play well on non-Windows platforms. Thanks for pointing out.

**Comment by foxcpp** [comment]
I guess we can merge this after https://github.com/Hexawolf/SSProto/pull/9 and https://github.com/Hexawolf/SSProto/pull/7. But if you think these PRs should be tested more (or something else is wrong) - feel free to merge without them.

**Comment by Hexawolf** [comment]
No more features will be added to SSProto 2 release. Merging branch dev into master.

**Pull request merged**